src/core/dynamic-user.c

```diff
@@ -452,7 +457,7 @@ static int dynamic_user_realize(DynamicUser *d, char **suggested_dirs, uid_t *re
         if (r < 0)
                 return r;
 
-        r = dynamic_user_pop(d, &uid, &uid_lock_fd);
+        r = dynamic_user_pop(d, &num, &uid_lock_fd);
         if (r < 0) {
```

Comment:
The variable gid is not set when `dynamic_user_pop()` succeeds. So, needs to add
```
if (r >= 0)
        gid = num;
else {
```